parasolr/query.py

```diff
@@ -123,6 +127,22 @@ def count(self) -> int:
         query_opts['rows'] = 0
         return self.solr.query(**query_opts, wrap=False)['response']['numFound']
 
+    def facet_fields(self) -> dict:
+        """Return a dictionary of facets and their values/counts (also as a
+        dicitionary).
+        """
+        if self._result_cache is not None:
+            # wrap to process facets and return as dictionary
+            # for Django template support
+            qr = QueryResponse(self._result_cache)
+            return dict(qr.facet_counts.facet_fields)
+        # since we just want a dictionary of facet fields, don't populate
+        # the result cache and do a wrapped request with an explicit cast
+        # to dict, no rows needed.
+        query_opts = self.query_opts()
+        query_opts['rows'] = 0
```

Comment:
I know you borrowed this from my code - I _think_ that kwargs passed in now take precedence over generated query opts, so it ought to be possible to just call `query` with the `rows=0` override and let it use all the other defaults.

It occurs to me the `count` query could probably drop faceting and highlighting options too...